### PR TITLE
Decorate documentation

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Changes include:
 + Added a page that describes the decorate pattern
 + Moved the installation instructions from the README to the site (this should help prevent issues like #485 in future)
 + Minor change to point series, it was not applying the container transform on enter

I'm still a little unsure about some of the terminology ... what do you call a selection that has been joined with data?

### CODE REVIEW

**Comment by chrisprice** [comment]
Sorry forgot about this one, it's got some merge conflicts - do you have time to fix them or do you want me to?

**Comment by ColinEberhardt** [comment]
> do you have time to fix them or do you want me to?

I'm happy to resolve the conflicts, I doubt there's anything that complicated. I'd appreciate any feedback on the 'decorate pattern' docs and restructuring.

File: site/src/_layouts/component.hbs [hunk 1]
```diff
@@ -37,6 +37,14 @@ layout: default
 
     <nav class="col-sm-3 hidden-xs">
       <ul id="sidebar" class="nav nav-stacked fixed">
+        <li>
```
Comment (chrisprice) [inline site/src/_layouts/component.hbs]: I know it's a bit of a corruption but any reason not to just do these as components and add them to the ```_config.yml```?

File: site/src/components/decorate-pattern.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        .select('circle')
+        .attr('r', 40);
+    });
+---
+
+Most chart APIs are complex and expansive in order to provide flexibility. With d3fc we have taken a different approach ...
+
+One of the most powerful features of D3 is the [data join](http://bost.ocks.org/mike/join/), where DOM elements are constructed via data-binding. Furthermore, the [general update pattern](http://bl.ocks.org/3808218) allows you to define exactly how these data-bound elements are modified as items are added, remove or updated.
+
+Unfortunately the [D3 component pattern](http://bost.ocks.org/mike/chart/), where re-useable units are encapsulated as functions, also encapsulates the data-join, hiding its power. For example, the [D3 axis component](https://github.com/mbostock/d3/wiki/SVG-Axes) uses a data-join internally, but as a consumer of this component you cannot add extra logic to the enter selection. The decorate pattern addresses this issue, by allowing the construction of component in such a way that their internal data join is 'exposed'.
```
Comment (chrisprice) [inline site/src/components/decorate-pattern.md]: component*s*

File: site/src/components/decorate-pattern.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+  {{{bar-example-code}}}
+
+  container.append('g')
+      .datum(data)
+      .call(bar);
+}());
+</script>
+
+Again, the enter selection is used and the required child element, in this case a `path` is selected. D3 selections allow you to specify values as functions of the bound data and index, here the index is used to cycle through a color scale.
```
Comment (chrisprice) [inline site/src/components/decorate-pattern.md]: > D3 selections allow you to

I'm not sure if that's quite right but I'm not sure of a better way of wording it.

File: site/src/components/decorate-pattern.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+    {{{enter-exit-example-code}}}
+
+      series.datum(data)
+        .transition()
+        .duration(500)
+        .call(point);
+  }
+
+  setInterval(render, 200);
```
Comment (chrisprice) [inline site/src/components/decorate-pattern.md]: requestAnimationFrame

File: site/src/style/variables.less [hunk 5]
```diff
@@ -18,3 +18,12 @@
 @jumbotron-color:       white;
 
 @section-vertical-padding: 30px;
+
```
Comment (chrisprice) [inline site/src/style/variables.less]: Do we need this?

**Comment by chrisprice** [comment]
Few comments about structure/grammar but content looks really good.

File: site/src/style/variables.less (hunk 5 above)
Comment (ColinEberhardt) [inline site/src/style/variables.less]: > Do we need this?

Subjective!

I just find that the standard bootstrap heading fonts lack enough distinction between H1 and H2:

screen shot 2015-08-05 at 21 34 59

I guess this is why people often vary other aspects of the font such as its weight or caps in order to distinguish between them.

File: site/src/components/decorate-pattern.md (hunk 4 above)
Comment (ColinEberhardt) [inline site/src/components/decorate-pattern.md]: > requestAnimationFrame

I don't really want 60 circles added per second, were you thinking of `requestAnimationFrame` plus something like `if (tick%10 === 0)`?

File: site/src/_layouts/component.hbs (hunk 1 above)
Comment (ColinEberhardt) [inline site/src/_layouts/component.hbs]: Hmmm ... I'll have a check and see how that pans out - although I do want one of them to be `index.html`.

**Comment by ColinEberhardt** [comment]
> Few comments about structure/grammar but content looks really good.

Thanks - I'll resolve the conflicts, fix typos and consider the other comments.

File: site/src/components/decorate-pattern.md (hunk 4 above)
Comment (chrisprice) [inline site/src/components/decorate-pattern.md]: Ah misunderstood what the code did, in that case there's not a lot of point.

File: site/src/style/variables.less (hunk 5 above)
Comment (chrisprice) [inline site/src/style/variables.less]: I think that's a reasonable progression and it's [standard enough](http://spencermortensen.com/articles/typographic-scale/). Maybe we could boost the base-font-size to make them more visually different?

Comment (ColinEberhardt) [inline site/src/style/variables.less]: Can't argue with the science!

I tried boosting the base font size, but funnily enough that just made everything bigger ;-) I'm happy to go with the standard.

**Pull request merged**